pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/UGICache.java

```diff
@@ -129,90 +135,93 @@ public void release(Entry timedProxyUGI, boolean forceClean) {
     }
 
     /**
-     * Iterate through all the entries in the queue for the given segment
+     * Iterate through all the entries in the queue for the given session's segment
      * and close expired {@link UserGroupInformation}, otherwise it resets
      * the timer for every non-expired entry.
      *
-     * @param segmentId
+     * @param session
      */
-    private void cleanup(Integer segmentId) {
+    private void cleanup(SessionId session) {
```

Comment:
The cleanup here is confusing because it is using the current session with any expired UGI